### PR TITLE
Add ability to connect to PostgreSQL via unix socket

### PR DESCRIPTION
**Pull request by vtrf**
This is related to #733 

- [x] read the [CONTRIBUTING guidelines](README.md#contributing)
- [x] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [ ] updated documentation if needed
- [x] updated CHANGELOG.md

### CODE REVIEW

**Comment by juanfont** [comment]
@vtrf would if be possible some details in https://github.com/juanfont/headscale/blob/main/config-example.yaml, to explain how to use this? I guess an extra line in line 116 clarifying that port should be blank for the socket, and host should be the path to it :)

**Comment by vtrf** [comment]
@juanfont Is this enough or should I go more in depth about the option?

File: config-example.yaml [hunk 1]
```diff
@@ -116,6 +116,9 @@ db_path: /var/lib/headscale/db.sqlite
 # # Postgres config
 # db_type: postgres
 # db_host: localhost
+
+# This should be set when using a TCP connection, set it to '0' or keep it unset if
+# connecting through a unix socket.
```
Comment (juanfont) [inline config-example.yaml:121]:
```suggestion
```

File: config-example.yaml [hunk 2]
```diff
@@ -116,6 +116,9 @@ db_path: /var/lib/headscale/db.sqlite
 # # Postgres config
 # db_type: postgres
 # db_host: localhost
+
+# This should be set when using a TCP connection, set it to '0' or keep it unset if
```
Comment (juanfont) [inline config-example.yaml:120]:
```suggestion
```

File: config-example.yaml [hunk 3]
```diff
@@ -116,6 +116,9 @@ db_path: /var/lib/headscale/db.sqlite
 # # Postgres config
```
Comment (juanfont) [inline config-example.yaml:116]:
```suggestion
# # Postgres config
# If using a Unix socket to connect to Postgres, set the socket path in the 'host' field and leave 'port' blank.
```

**Comment by juanfont** [comment]
@vtrf can you have a look at the suggestions?

**Review by juanfont — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**